sql/log_event.cc

```diff
@@ -12171,6 +12197,24 @@ void Ignorable_log_event::print(FILE *,
 }
 #endif
 
+ulonglong Rows_query_log_event::extract_last_timestamp() const {
+  boost::property_tree::ptree pt;
+  std::string json = extract_trx_meta_data();
+  if (json.empty()) return 0;
+  std::istringstream is(json);
+  try {
+    read_json(is, pt);
+  } catch (const std::exception &e) {
+    sql_print_error("Error while parsing metadata JSON for timestamps");
+    return 0;
+  }
+
+  auto timestamps = pt.get_child("ts", boost::property_tree::ptree());
```

Comment:
There is also `ptree::get_child_optional()` which does not insert default value.

Reply:
Frot the boost doc: "Get the child at the given path, or return @p default_value." - this doesn't insert anything, it only returns the requested part or the right side. (which is why it returns a const&)